force-app/main/default/classes/ClassesObjectsMethods.cls

```diff
@@ -228,18 +298,39 @@ public with sharing class ClassesObjectsMethods {
     public class Book {
         // Declare three public instance variables - title, author, pages
         // title
+        public String title;
         // author
+        public String author;
         // pages
+        public Integer pages;
     
         // Book constructor to initialize the title and author
-        public Book() {
-            // Initialize the title and author
+        public Book(String title, String author, Integer pages) {
+        // Initialize the title and author
+        this.title = title;
+        this.author = author;
+        this.pages = pages;
+
         }
     
         // Method that returns the details of the book
         public String getBookDetails() {
-            // return A string containing the details of the book in the format "Title: <title>, Author: <author>"
-            return null; // Replace null with the variable you used to store the result
+        // return A string containing the details of the book in the format "Title: <title>, Author: <author>"
+        return 'Title: ' + title + ', Author: ' + author; // Replace null with the variable you used to store the result
+        }
+        // Getter method for title
+        public String getTitle() {
+        return this.title;
+        }
+
+        // Getter method for author
+        public String getAuthor() {
+        return this.author;
+        }
+
+        // Getter method for pages
+        public Integer getPages() {
+        return this.pages;
```

Comment:
You only really need a getter if you are making the properties private. otherwise you can call them directly with dot notation.